client/src/unifyfs.c

```diff
@@ -276,6 +279,20 @@ inline int unifyfs_stack_unlock(void)
     return 0;
 }
 
+inline void unifyfs_normalize_path(const char* path, char* normalized)
+{
+    /* if we have a relative path, prepend the current working directory */
+    if (path[0] != '/' && unifyfs_cwd != NULL) {
+        /* got a relative path, add our cwd */
+        snprintf(normalized, UNIFYFS_MAX_FILENAME, "%s/%s", unifyfs_cwd, path);
+    } else {
+        snprintf(normalized, UNIFYFS_MAX_FILENAME, "%s", path);
+    }
+
+    /* TODO: normalize path to handle '.', '..',
+     * and extra or trailing '/' characters */
```

Comment:
Would `realpath()` help here?

Reply:
...ah maybe not if it does real file lookups...

Reply:
Right, we need something like realpath but that operates on unifyfs paths.  Is there something available that works for path strings that don't actually exist?

Reply:
Not that I know of :disappointed: